src/command/cmd_funcs.c

```diff
@@ -8606,24 +8668,37 @@ _update_presence(const resource_presence_t resource_presence,
     }
 }
 
-// helper function for boolean preference commands
+/**
+ * Sets a boolean preference based on the provided argument.
+ *
+ * @param arg     The argument value specifying the preference state ("on" or "off").
+ * @param command The name of the command being executed.
+ * @param display The display name of the preference being set.
+ * @param pref    The preference identifier to be set.
+ *
+ * @return TRUE if the preference was successfully set, FALSE otherwise.
+ */
 static gboolean
 _cmd_set_boolean_preference(gchar* arg, const char* const command,
                             const char* const display, preference_t pref)
 {
+    gboolean prev_state = prefs_get_boolean(pref);
     if (arg == NULL) {
-        cons_bad_cmd_usage(command);
+        cons_show("%s is %s.", display, prev_state ? "enabled" : "disabled");
         return FALSE;
-    } else if (g_strcmp0(arg, "on") == 0) {
-        cons_show("%s enabled.", display);
+    }
+
+    if (g_ascii_strcasecmp(arg, "on") == 0) {
+        cons_show("%s %senabled.", display, prev_state ? "is already " : "");
         prefs_set_boolean(pref, TRUE);
-    } else if (g_strcmp0(arg, "off") == 0) {
-        cons_show("%s disabled.", display);
+    } else if (g_ascii_strcasecmp(arg, "off") == 0) {
+        cons_show("%s %sdisabled.", display, !prev_state ? "is already " : "");
```

Comment:
@jubalh prefers case sensitivity for arguments: https://github.com/profanity-im/profanity/pull/1780#discussion_r1039389092 :shrug: